tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -26,9 +26,70 @@ use tracing_core::{
     Metadata,
 };
 
-/// A `Layer` which filters spans and events based on a set of filter
+/// A [`Layer`] which filters spans and events based on a set of filter
 /// directives.
-// TODO(eliza): document filter directive syntax?
+///
+/// # Directives
+///
+/// A filter consists of the following:
+/// - One or more directives which match on [`Span`]s and [`Event`]s.
+/// - A maximum verbosity [`level`] which enables (e.g., _selects for_)
+///   spans and events that match. Like `log`, `tracing` considers less exclusive
+///   levels (like `trace` or `info`) to be more verbose than more exclusive levels
+///   (like `error` or `warn`).
```

Comment:
I would say the levels are _part_ of a directive; each directive sets its own level filter